src/main/java/hudson/plugins/jira/JiraIssueFieldUpdater.java

```diff
@@ -0,0 +1,186 @@
+        return this.field_value;
+    }
+
+    @DataBoundSetter
+    public void setFieldValue(String field_value) {
+        this.field_value = field_value;
+    }
+
+    @DataBoundConstructor
+    public JiraIssueFieldUpdater(AbstractIssueSelector issueSelector, String field_id, String field_value) {
+        this.issueSelector = issueSelector;
+        this.field_id = field_id;
+        this.field_value = field_value;
+    }
+
+    public String prepare_field_id(String field_id) {
+        String prepared = field_id;
+        if (!prepared.startsWith("customfield_"))
+            prepared = "customfield_" + prepared;
+        return prepared;
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
+            throws InterruptedException, IOException {
+        PrintStream logger = listener.getLogger();
+
+        AbstractIssueSelector selector = issueSelector;
+        if (selector == null) {
+            logger.println("[JIRA][JiraIssueFieldUpdater] No issue selector found!");
+            throw new IOException("[JIRA][JiraIssueFieldUpdater] No issue selector found!");
+        }
+
+        JiraSite site = JiraSite.get(run.getParent());
+        if (site == null) {
+            logger.println(Messages.Updater_NoJiraSite());
+            run.setResult(Result.FAILURE);
+            return;
+        }
+
+        JiraSession session = null;
+        try {
+            session = site.getSession();
+        } catch (IOException e) {
+            listener.getLogger().println(Messages.Updater_FailedToConnect());
+            e.printStackTrace(listener.getLogger());
+        }
+        if (session == null) {
+            logger.println(Messages.Updater_NoRemoteAccess());
+            run.setResult(Result.FAILURE);
+            return;
+        }
+
+        Set<String> issues = selector.findIssueIds(run, site, listener);
+        if (issues.isEmpty()) {
+            logger.println("[JIRA][JiraIssueFieldUpdater] Issue list is empty!");
+            return;
+        }
+
+        List<JiraIssueField> fields = Lists.newArrayList();
+        fields.add(new JiraIssueField(prepare_field_id(field_id), field_value));
+
+        for (String issue : issues) {
+            submitFeilds(session, issue, fields, logger);
+        }
+    }
+
+    public void submitFeilds(JiraSession session, String issueId, List<JiraIssueField> fields, PrintStream logger) {
+        try {
+            session.addFields(issueId, fields);
+        } catch (RestClientException e) {
+
+            if (e.getStatusCode().or(0).equals(404)) {
+                logger.println(issueId + " - JIRA issue not found. Dropping comment from update queue.");
+            }
+
+            if (e.getStatusCode().or(0).equals(403)) {
+                logger.println(issueId
+                        + " - Jenkins JIRA user does not have permissions to comment on this issue. Preserving comment for future update.");
```

Comment:
in Updater the failed issues are preserved by:
https://github.com/jenkinsci/jira-plugin/blob/master/src/main/java/hudson/plugins/jira/Updater.java#L120

Here this logic is missing:
https://github.com/tequillaz/jira-plugin/blob/cd2d9db29b369df80f5ddc61a465765172042c3e/src/main/java/hudson/plugins/jira/JiraIssueFieldUpdater.java#L131